src/app/services/country.service.ts

```diff
@@ -1,49 +1,53 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable, of, map, shareReplay, catchError } from 'rxjs';
+import { Observable, of, catchError, map } from 'rxjs';
 import { Country, CountryDetail } from '../models/country.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CountryService {
-  private apiUrl = 'https://restcountries.com/v3.1/all';
-  private searchByNameApiUrl = 'https://restcountries.com/v3.1/name'; // API for search by name
-  private searchByCapitalApiUrl = 'https://restcountries.com/v3.1/capital'; // API for search by capital
+  private apiUrl = 'https://restcountries.com/v3.1';
 
   constructor(private http: HttpClient) {}
   // Get all countries from cached API call
   getAllCountries() {
-    return this.http.get<any>(this.apiUrl);
+    const url = `${this.apiUrl}/all`;
+
+    return this.http.get<any>(url);
   }
 
   // Get specific country by name from the cached data
-  getCountryByName(name: string): Observable<CountryDetail | any> {
-    return this.http
-      .get<any>(this.apiUrl)
-      .pipe(
-        map(
-          (countries) =>
-            countries.find((country: any) => country.name.common === name) ||
-            null
-        )
-      );
+  getCountryByName(name: string): any {
   }
-
   // Search countries using different APIs (by name or capital)
   searchCountries(
     searchTerm: string,
-    searchBy: 'name' | 'capital'
+    searchBy: 'name' | 'capital' | 'region' | 'alpha'
   ): Observable<Country[]> {
     if (!searchTerm.trim()) {
-      return this.getAllCountries();
+      return this.getAllCountries(); // Return all countries if the search term is empty
```

Comment:
This part is ideally handled inside the effect itself. Please remove it from here.